cvmfs/swissknife_main.cc

```diff
@@ -187,10 +197,88 @@ int main(int argc, char **argv) {
+    string files_removed = command->statistics()->
+                     Lookup("Publish.n_files_removed")->ToString();
+    string files_changed = command->statistics()->
+                     Lookup("Publish.n_files_changed")->ToString();
+    string dir_added = command->statistics()->
+                         Lookup("Publish.n_directories_added")->ToString();
+    string dir_removed = command->statistics()->
+                     Lookup("Publish.n_directories_removed")->ToString();
+    string dir_changed = command->statistics()->
+                     Lookup("Publish.n_directories_changed")->ToString();
+    string bytes_added = command->statistics()->
+                     Lookup("Publish.sz_added_bytes")->ToString();
+    string bytes_removed = command->statistics()->
+                     Lookup("Publish.sz_removed_bytes")->ToString();
+
+    struct tm time_ptr;
+    char date_and_time[50];
+    time_t t = time(NULL);
+    gmtime_r(&t, &time_ptr);      // take UTC
+    // timestamp format
+    strftime(date_and_time, 1000, "%Y-%m-%d %H:%M:%S", &time_ptr);
+    string timestamp_value(date_and_time);
+
+    string insert_statement = "INSERT INTO publish_statistics ("
+                              "timestamp,"
+                              "files_added,"
+                              "files_removed,"
+                              "files_changed,"
+                              "directories_added,"
+                              "directories_removed,"
+                              "directories_changed,"
+                              "sz_bytes_added,"
+                              "sz_bytes_removed)"
+                              " VALUES("
+                              "'"+timestamp_value+"',"+  // TEXT datatype
+                              files_added+"," +
+                              files_removed +","+
+                              files_changed + "," +
+                              dir_added + "," +
+                              dir_removed + "," +
+                              dir_changed + "," +
+                              bytes_added + "," +
+                              bytes_removed + ");";
+    LogCvmfs(kLogCvmfs, kLogStdout, "%s", insert_statement.c_str());
+    sqlite::Sql insert(db->sqlite_db(), insert_statement);
+
+    if (!db->BeginTransaction()) {
+      LogCvmfs(kLogCvmfs, kLogStdout, "BeginTransaction failed!");
+    }
+
+    if (!insert.Execute()) {
+      LogCvmfs(kLogCvmfs, kLogStdout, "insert.Execute failed!");
+    }
+
+    if (!insert.Reset()) {
+      LogCvmfs(kLogCvmfs, kLogStdout, "insert.Reset() failed!");
+    }
+
+    if (!db->CommitTransaction()) {
+      LogCvmfs(kLogCvmfs, kLogStdout, "CommitTransaction failed!");
+    }
+
     LogCvmfs(kLogCvmfs, kLogStdout, "Command statistics");
     LogCvmfs(kLogCvmfs, kLogStdout, "%s",
              command->statistics()
              ->PrintList(perf::Statistics::kPrintHeader).c_str());
+    // delete db;  // uncomment this to fail ?
```

Comment:
If you look in `sql_impl.h:150`, there is an error when closing the db.
Try to run the command with CVMFS_SERVER_DEBUG=3 to get some debug output from inside the `DatabaseRaiiWrapper::Close()` method.

Some docs about closing SQLite dbs: https://sqlite.org/c3ref/close.html